src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -0,0 +1,114 @@
+import com.breskul.bring.exceptions.NuSuchBeanConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Standalone application context, accepting component classes as input — in particular @Configuration-annotated classes,
+ * but also plain @Component types
+ */
+public class AnnotationConfigApplicationContext implements ApplicationContext {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BringApplication.class);
+    private final Map<String, Object> context = new ConcurrentHashMap<>();
+    private final String packagePath;
+
+    public AnnotationConfigApplicationContext(String packageName) {
+        Objects.requireNonNull(packageName);
+        this.packagePath = packageName;
+        scan();
+    }
+
+    /**
+     * Scanning a package for classes or methods annotated by @Component and @Bean
+     */
+    private void scan() {
+    }
+
+    /**
+     * Store bean in the context
+     *
+     * @param beanName The name of bean
+     * @param bean     The instance of bean
+     * @param <T>      The generic type of instance
+     */
+    private <T> void storeBean(String beanName, T bean) {
+        Objects.requireNonNull(beanName);
+        Objects.requireNonNull(bean);
+        LOGGER.info("Bring Framework: Creating bean >>> {}: {}", beanName, bean);
+        context.put(beanName, bean);
+    }
+
+    /**
+     * Maps filter by bean type
+     *
+     * @param beanType The Class of type
+     * @param <T>      type of the bean
+     * @return Predicate - filter isAssignableFrom for values types
+     */
+    private <T> Predicate<Map.Entry<String, Object>> filterByBeanType(Class<T> beanType) {
+        return (entry) -> entry.getValue().getClass().isAssignableFrom(beanType);
+    }
+
+    /**
+     * Creating an instance of bean by beanType
+     *
+     * @param beanType - The type of bean instance
+     * @param <T>
+     * @return instance of the created bean
+     */
+    private <T> T createInstance(Class<T> beanType) {
+        var constructor = Arrays.stream(beanType.getConstructors())
+                .findFirst()
+                .orElseThrow(() -> {
+                    throw new NuSuchBeanConstructor("Bean type: " + beanType.getName());
+                });
+        try {
+            return beanType.cast(constructor.newInstance());
+        } catch (Exception ex) {
+            throw new BeanInitializingException("Bean type: " + beanType.getName());
+        }
+    }
+
+    @Override
+    public <T> T getBean(Class<T> beanType) throws NoSuchBeanException, NoUniqueBeanException {
+        Map<String, T> beans = getAllBeans(beanType);
+        if (beans.size() > 1) {
+            throw new NoUniqueBeanException("Bean type: " + beanType.getName());
```

Comment:
suggested change but I'll approve anyway
`			throw new NoUniqueBeanException("There are %s beans found for %s, expected 1.".formatted(beans.size(), beanType.getName()));
`